cmds/core/cp/cp.go

```diff
@@ -46,17 +50,17 @@ func init() {
 		os.Args[0] = "cp [-wRrifvP] file[s] ... dest"
 		defUsage()
 	}
-	flag.BoolVarP(&flags.recursive, "RECURSIVE", "R", false, "copy file hierarchies")
-	flag.BoolVarP(&flags.recursive, "recursive", "r", false, "alias to -R recursive mode")
-	flag.BoolVarP(&flags.ask, "interactive", "i", false, "prompt about overwriting file")
-	flag.BoolVarP(&flags.force, "force", "f", false, "force overwrite files")
-	flag.BoolVarP(&flags.verbose, "verbose", "v", false, "verbose copy mode")
-	flag.BoolVarP(&flags.noFollowSymlinks, "no-dereference", "P", false, "don't follow symlinks")
+	flag.BoolVarP(&f.recursive, "RECURSIVE", "R", false, "copy file hierarchies")
+	flag.BoolVarP(&f.recursive, "recursive", "r", false, "alias to -R recursive mode")
+	flag.BoolVarP(&f.ask, "interactive", "i", false, "prompt about overwriting file")
+	flag.BoolVarP(&f.force, "force", "f", false, "force overwrite files")
+	flag.BoolVarP(&f.verbose, "verbose", "v", false, "verbose copy mode")
+	flag.BoolVarP(&f.noFollowSymlinks, "no-dereference", "P", false, "don't follow symlinks")
 }
 
 // promptOverwrite ask if the user wants overwrite file
-func promptOverwrite(dst string) (bool, error) {
-	fmt.Printf("cp: overwrite %q? ", dst)
+func promptOverwrite(dst string, out io.Writer) (bool, error) {
```

Comment:
Should input be an io.Reader parameter instead of a global?

Reply:
Thus this is called in `setupPreCallback` and this needs an io.Writer - is it feasible to make it io.Reader? And what do you mean with global?